cfpb/cfgov-refresh
# P4C: Front end fixes

Fixes JS error that prevents selection of a school when one or more of its rate values are missing. Reduces calls to `history.replaceState` to avoid exceeding Safari limits.


## Changes

- In `chart-view.js`, check whether a school cohort is available before attempting to get its percentile rank. This avoids a JS error that was preventing the selection of a few schools.
- In `update-models,js`, update the state to reflect missing rate values so their chart sections can be hidden on the school comparison view.
- Only update the URL / call `history.replaceState` from the state model when one of the url params changes
- Reduce unnecessary calls to update the URL / replaceState while setting model values from the URL


## How to test this PR

1. Check out branch
2. Run gulp
3. Navigate to [college costs](http://localhost:8000/paying-for-college/your-financial-path-to-graduation) in Safari and verify that the app works as expected
4.  Midwestern University: [Navigate to app](http://localhost:8000/paying-for-college/your-financial-path-to-graduation/?iped=143853&houp=onCampus&typp=bachelors&lenp=4&depp=dependent&cobs=n&houx=826&fdx=340&clhx=87&trnx=324&hltx=121&entx=94&retx=22&taxx=-6&othx=385). Verify that Midwestern University is selected and that only the default rate chart is displayed on the `Worth your investment` > `Compare this school to others` tab.
5. Oxnard: [Navigate to app](http://localhost:8000/paying-for-college/your-financial-path-to-graduation/?iped=120421&houp=onCampus&typp=bachelors&lenp=4&ratp=inState&depp=dependent&cobs=n&houx=826&fdx=340&clhx=87&trnx=324&hltx=121&entx=94&retx=22&taxx=-6&othx=385). Verify that Oxnard shows in school field and that only the Grad and Default rate sections are displayed on the school comparison tab


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/cfgov-refresh/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing

#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
